evm/src/keccak_sponge/keccak_sponge_stark.rs

```diff
@@ -211,7 +211,7 @@ impl<F: RichField + Extendable<D>, const D: usize> KeccakSpongeStark<F, D> {
         operations: Vec<KeccakSpongeOp>,
         min_rows: usize,
     ) -> Vec<[F; NUM_KECCAK_SPONGE_COLUMNS]> {
-        let mut rows = vec![];
+        let mut rows = Vec::with_capacity(operations.len().max(min_rows).next_power_of_two());
```

Comment:
Doesn't `generate_rows_for_op` typically produce multiple rows per operation? So isn't this value off from the final vector size by that factor?

Reply:
Hmm good point, this may still cause reallocation. I guess I should change it to something like

```suggestion
        let base_len: usize = operations
            .iter()
            .map(|op| op.input.len() / KECCAK_RATE_BYTES + 1)
            .sum();
        let mut rows = Vec::with_capacity(base_len.max(min_rows).next_power_of_two());
```

Reply:
Yes, that looks good!

Reply:
I've addressed it in follow-up commits, as well as Daniel's comment.

Reply:
And related to this point, the allocation of the `HashMap` in `get_sigma_map()` at plonky2/src/plonk/permutation_argument.rs is not ideal, as it may still reallocate, but given the usually huge count of distinct keys, it saves quite a bit of reallocations still.

Reply:
@npwardberkeley Was there anything else I should deal with related to this PR?

Reply:
Oh sorry no, I think it looks good!